tests/integration/local/start_lambda/test_start_lambda_terraform_applications.py

```diff
@@ -228,6 +246,64 @@ def test_invoke_function(self, function_name, expected_output):
         self.assertEqual(response.get("StatusCode"), 200)
 
 
+@skipIf(
+    not CI_OVERRIDE,
+    "Skip Terraform test cases unless running in CI",
+)
+class TestInvalidTerraformApplicationThatReferToS3BucketNotCreatedYet(StartLambdaTerraformApplicationIntegBase):
+    terraform_application = "/testdata/invoke/terraform/invalid_no_local_code_project"
+    template_path = None
+    hook_package_id = "terraform"
+    beta_features = True
+
+    @classmethod
+    def setUpClass(cls):
+        # over write the parent setup
+        pass
+
+    @classmethod
+    def tearDownClass(cls):
+        # over write the parent tear down
+        pass
+
+    def tearDown(self):
+        # delete the override file
+        try:
+            shutil.rmtree(str(Path(self.working_dir).joinpath(".aws-sam-iacs")))
+            shutil.rmtree(str(Path(self.working_dir).joinpath(".terraform")))
+            os.remove(str(Path(self.working_dir).joinpath(".terraform.lock.hcl")))
+        except FileNotFoundError:
+            pass
+
+    @pytest.mark.flaky(reruns=3)
+    def test_invoke_function(self):
+        self.working_dir = self.integration_dir + self.terraform_application
```

Comment:
Should we consider moving the directory and/or docker setup to `setUp()` or a helper function? There is only one test for this class and I'm not sure if we even need other tests for this case, but just something to keep in mind